frameworks/java/prewarm.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+echo "loading" > /workspace/prewarm.status
+
+# prewarm by starting the gradle daemon. Running an initial test build will also speed things up a bit
+cd /runner/frameworks/java && gradle --daemon --offline test
```

Comment:
It was pretty bad. I think things went from an avg of 8 seconds down to 3 with Leiningen. We used to get a lot of users complaining about Java run times, which is what prompted us to try to find a better way. 

Now with Gradle it's more like 4 seconds, but also there is an initial 6 - 8 second warmup time - which shouldn't affect most users unless they happen to hit a container twice in a row before it is fully warmed up. 

Gradle obviously isn't being introduced for performance reasons but instead for extended java capabilities. We have a bunch of enterprise clients who need Spring support, and we eventually want to expand to supporting Android, which needs Gradle. Of course CW should be able to benefit from this as well - though framework challenges haven't become popular on there yet.

If this ends up causing performance issues then I may roll back some of these changes, but I wanted to see how well it runs on our staging cluster first. If things don't go well then we may have to prioritize persistent containers even higher than it is now.

> `docker run --rm -i --memory=1GB`

We never specify dedicated memory limits when starting containers, so they may all be competing with each other. Java is the only image that actually pre-warms anything, the others just sit there - so it might now suck up all of the memory that normally was left open before.